src/tracks/Pattern.cpp

```diff
@@ -421,6 +421,32 @@ void Pattern::loadSettings( const QDomElement & _this )
 
 
 
+Pattern *  Pattern::previousPattern() const
+{
+	return adjacentPatternByOffset(-1);
+}
+
+
+
+
+Pattern *  Pattern::nextPattern() const
+{
+	return adjacentPatternByOffset(1);
+}
+
+
+
+
+Pattern * Pattern::adjacentPatternByOffset(int offset) const
+{
+	QVector<TrackContentObject *> tcos = m_instrumentTrack->getTCOs();
+	int tcoNum = m_instrumentTrack->getTCONum(this);
+	return dynamic_cast<Pattern*>(tcos.value(tcoNum + offset, NULL));
+}
```

Comment:
@tresf `tcos.value` will do bounds checking, if the index is out of bounds it returns NULL. It's the responsibility of the click handler to check for NULL.